_posts/2020-04-18-schedule-your-functions.md

```diff
@@ -0,0 +1,130 @@
+---
+title: "On time - schedule your OpenFaaS functions"
+description: "In this post Martin will show you how to deploy a component which will execute your functions on a schedule"
+date: 2020-04-18
+image: /images/2020-04-18-schedule-your-functions/accuracy-alarm-clock-analogue.jpg
+categories:
+  - cronjobs
+  - kubernetes
+  - serverless
+  - linux
+  - tutorial
+  - example
+author_staff_member: martin
+dark_background: true
+
+---
+
+In this blog post I will show you how you can run your functions on schedule. We will first set our environment and then we will run the cron-connector with example function which will be executed on a schedule.
+
+One of the things we need very often is our functions to be executed on a schedule. This is where the cron-connector shines. We will showcase how you can use it in this post.
+
+## Motivation
+
+OpenFaaS provides very convenient and flexible way to quickly write and expose your code to be used by external systems. Often times those systems don't have a hook to which you can schedule function execution. Also often times we need to run those functions without external systems, we just want our code to be ran on an interval.
+
+For those purposes [connector-sdk](https://github.com/openfaas-incubator/connector-sdk) was created. It provides the means to quickly adapt different systems to invoke functions. Some of them are [Kafka](https://github.com/openfaas-incubator/kafka-connector), [NATS](https://github.com/openfaas-incubator/nats-connector), [AWS SNS](https://github.com/affix/OpenFaaS-SNS) and [many more](https://docs.openfaas.com/reference/triggers/).
+
+Now we need our functions to to be invoked on an interval. The [cron-connector](https://github.com/openfaas-incubator/cron-connector) exists for those purposes much like Kubernetes provides specific object called `CronJob` which executes code on a schedule or how in Linux we have cron configuration which run scripts on a schedule.
```

Comment:
It would be good to connect this to the previous paragraph by saying that the cron-connector is another example of the connector-sdk.  the name implies it but, it might be good to make that explicit.  I might write it something like this 

```suggestion
Now we need our functions to to be invoked on an interval. In Linux we have the [cron configuration](https://en.wikipedia.org/wiki/Cron) that allows us to execute gode on a schedule.  In Kubernetes, this was abstracted into the `CronJob` object.  For any OpenFaaS cluster, the [cron-connector](https://github.com/openfaas-incubator/cron-connector) uses the `connector-sdk` to provide the same functionality, running our functions on a predefined schedule.
```

Reply:
Yep, very good suggestion will change as proposed :+1:

Reply:
Done :heavy_check_mark: